### PR TITLE
Replace deprecated TensorFlow functions

### PR DESCRIPTION
**Pull request by asnelt**
The functions histogram_summary, image_summary, scalar_summary, merge_summary and SummaryWriter are deprecated. The change replaces them with tf.summary.histogram, tf.summary.image, tf.summary.scalar, tf.summary.merge and tf.summary.FileWriter. Also, the argument name "labels" is replaced with "targets" in sigmoid_cross_entropy_with_logits and y[i] is explicitely cast to integer to address all warnings.

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def visualize(sess, dcgan, config, option):
     save_images(samples, [8, 8], './samples/test_%s.png' % strftime("%Y-%m-%d %H:%M:%S", gmtime()))
   elif option == 1:
     values = np.arange(0, 1, 1./config.batch_size)
-    for idx in xrange(100):
```
Comment (ppwwyyxx) [inline utils.py]: I'll suggest using six.moves.range for "actual" compatibility between python 2 and 3.

**Comment by carpedm20** [comment]
@asnelt If you suggest to change summary ops, why don't you delete [these try catch code](https://github.com/asnelt/DCGAN-tensorflow/blob/095cf88a84a76f89eaf578c2aff2ca3e18462cfe/ops.py#L9)? Is there any special reason?

**Comment by asnelt** [comment]
Ah, I am sorry for the noise.
